Failing GCP Test - `TestGCP_CollectMetrics/two_collectors_with_no_errors`

Test is failing on the main branch:

```shell
--- FAIL: TestGCP_CollectMetrics (0.00s)
    --- FAIL: TestGCP_CollectMetrics/two_collectors_with_no_errors (0.00s)
        gcp_test.go:206: 
                Error Trace:    /Users/loganballard/projects/cloudcost-exporter/pkg/google/gcp_test.go:206
                Error:          Not equal: 
                                expected: &utils.MetricResult{FqName:"cloudcost_exporter_collector_last_scrape_duration_seconds", Labels:utils.LabelMap{"collector":"test", "provider":"gcp"}, Value:0, MetricType:2}
                                actual  : &utils.MetricResult{FqName:"cloudcost_exporter_collector_last_scrape_error", Labels:utils.LabelMap{"collector":"test", "provider":"gcp"}, Value:0, MetricType:2}
                            
                                Diff:
                                --- Expected
                                +++ Actual
                                @@ -1,3 +1,3 @@
                                 (*utils.MetricResult)({
                                - FqName: (string) (len=57) "cloudcost_exporter_collector_last_scrape_duration_seconds",
                                + FqName: (string) (len=46) "cloudcost_exporter_collector_last_scrape_error",
                                  Labels: (utils.LabelMap) (len=2) {
                Test:           TestGCP_CollectMetrics/two_collectors_with_no_errors
FAIL
FAIL    github.com/grafana/cloudcost-exporter/pkg/google        0.304s
```

@Pokom left a comment here

```go
				// TODO: @pokom need to figure out why _sometimes_ this fails if we set it to *.Times(tt.numCollectors)

```

--- a/pkg/google/gcp_test.go
+++ b/pkg/google/gcp_test.go
@@ -162,18 +162,25 @@ func TestGCP_CollectMetrics(t *testing.T) {
 			wg.Done()
 
 			wg.Wait()
-			ignoredMetricSuffix := []string{"duration_seconds", "last_scrape_time"}
 			var metrics []*utils.MetricResult
-			// I don't love using a named loop, but this allows the inner loop to properly continue if the condition has been met.
-		metricsLoop:
-			for m := range ch {
-				metric := utils.ReadMetrics(m)
+			var ignoreMetric = func(metricName string) bool {
+				ignoredMetricSuffix := []string{
+					"duration_seconds",
+					"last_scrape_time",
+				}
 				for _, suffix := range ignoredMetricSuffix {
-					if strings.Contains(metric.FqName, suffix) {
-						continue metricsLoop
+					if strings.Contains(metricName, suffix) {
+						return true
 					}
 				}
 
+				return false
+			}
+			for m := range ch {
+				metric := utils.ReadMetrics(m)
+				if ignoreMetric(metric.FqName) {
+					continue
+				}
 				metrics = append(metrics, metric)
 			}
 			assert.ElementsMatch(t, metrics, tt.expectedMetrics)

--- a/pkg/google/gcp_test.go
+++ b/pkg/google/gcp_test.go
@@ -75,6 +75,14 @@ func TestGCP_CollectMetrics(t *testing.T) {
 	}{
 		"no error if no collectors": {
 			numCollectors: 0,
+			expectedMetrics: []*utils.MetricResult{
+				{
+					FqName:     "cloudcost_exporter_last_scrape_error",
+					Labels:     utils.LabelMap{"provider": "gcp"},
+					Value:      0,
+					MetricType: prometheus.GaugeValue,
+				},
+			},
 		},
 		"bubble-up single collector error": {
 			numCollectors: 1,
@@ -86,6 +94,12 @@ func TestGCP_CollectMetrics(t *testing.T) {
 					FqName:     "cloudcost_exporter_collector_last_scrape_error",
 					Labels:     utils.LabelMap{"provider": "gcp", "collector": "test"},
 					Value:      1,
+					MetricType: prometheus.GaugeValue,
+				},
+				{
+					FqName:     "cloudcost_exporter_last_scrape_error",
+					Labels:     utils.LabelMap{"provider": "gcp"},
+					Value:      0,
 					MetricType: prometheus.GaugeValue,
 				},
 			},
@@ -100,7 +114,18 @@ func TestGCP_CollectMetrics(t *testing.T) {
 					Value:      0,
 					MetricType: prometheus.GaugeValue,
 				},
-			},
+				{
+					FqName:     "cloudcost_exporter_collector_last_scrape_error",
+					Labels:     utils.LabelMap{"provider": "gcp", "collector": "test"},
+					Value:      0,
+					MetricType: prometheus.GaugeValue,
+				},
+				{
+					FqName:     "cloudcost_exporter_last_scrape_error",
+					Labels:     utils.LabelMap{"provider": "gcp"},
+					Value:      0,
+					MetricType: prometheus.GaugeValue,
+				}},
 		},
 	}
 	for name, tt := range tests {

--- a/pkg/google/gcp_test.go
+++ b/pkg/google/gcp_test.go
@@ -9,6 +9,7 @@ import (
 	"testing"
 
 	"github.com/prometheus/client_golang/prometheus"
+	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 	"go.uber.org/mock/gomock"
 
@@ -87,30 +88,6 @@ func TestGCP_CollectMetrics(t *testing.T) {
 					Value:      1,
 					MetricType: prometheus.GaugeValue,
 				},
-				{
-					FqName:     "cloudcost_exporter_collector_last_scrape_duration_seconds",
-					Labels:     utils.LabelMap{"provider": "gcp", "collector": "test"},
-					Value:      0,
-					MetricType: prometheus.GaugeValue,
-				},
-				{
-					FqName:     "cloudcost_exporter_collector_last_scrape_time",
-					Labels:     utils.LabelMap{"provider": "gcp", "collector": "test"},
-					Value:      0,
-					MetricType: prometheus.GaugeValue,
-				},
-				{
-					FqName:     "cloudcost_exporter_last_scrape_error",
-					Labels:     utils.LabelMap{"provider": "gcp"},
-					Value:      0,
-					MetricType: prometheus.GaugeValue,
-				},
-				{
-					FqName:     "cloudcost_exporter_last_scrape_duration_seconds",
-					Labels:     utils.LabelMap{"provider": "gcp"},
-					Value:      1,
-					MetricType: prometheus.GaugeValue,
-				},
 			},
 		},
 		"two collectors with no errors": {
@@ -120,42 +97,6 @@ func TestGCP_CollectMetrics(t *testing.T) {
 				{
 					FqName:     "cloudcost_exporter_collector_last_scrape_error",
 					Labels:     utils.LabelMap{"provider": "gcp", "collector": "test"},
-					Value:      0,
-					MetricType: prometheus.GaugeValue,
-				}, {
-					FqName:     "cloudcost_exporter_collector_last_scrape_duration_seconds",
-					Labels:     utils.LabelMap{"provider": "gcp", "collector": "test"},
-					Value:      0,
-					MetricType: prometheus.GaugeValue,
-				}, {
-					FqName:     "cloudcost_exporter_collector_last_scrape_time",
-					Labels:     utils.LabelMap{"provider": "gcp", "collector": "test"},
-					Value:      0,
-					MetricType: prometheus.GaugeValue,
-				},
-				{
-					FqName:     "cloudcost_exporter_collector_last_scrape_error",
-					Labels:     utils.LabelMap{"provider": "gcp", "collector": "test"},
-					Value:      0,
-					MetricType: prometheus.GaugeValue,
-				},
-				{
-					FqName:     "cloudcost_exporter_collector_last_scrape_duration_seconds",
-					Labels:     utils.LabelMap{"provider": "gcp", "collector": "test"},
-					Value:      0,
-					MetricType: prometheus.GaugeValue,
-				},
-
-				{
-					FqName:     "cloudcost_exporter_collector_last_scrape_time",
-					Labels:     utils.LabelMap{"provider": "gcp", "collector": "test"},
-					Value:      0,
-					MetricType: prometheus.GaugeValue,
-				},
-
-				{
-					FqName:     "cloudcost_exporter_last_scrape_error",
-					Labels:     utils.LabelMap{"provider": "gcp"},
 					Value:      0,
 					MetricType: prometheus.GaugeValue,
 				},
@@ -197,21 +138,20 @@ func TestGCP_CollectMetrics(t *testing.T) {
 
 			wg.Wait()
 			ignoredMetricSuffix := []string{"duration_seconds", "last_scrape_time"}
+			var metrics []*utils.MetricResult
 			// I don't love using a named loop, but this allows the inner loop to properly continue if the condition has been met.
 		metricsLoop:
-			for _, expectedMetric := range tt.expectedMetrics {
-				metric := utils.ReadMetrics(<-ch)
-				// We don't care about the value for the scrape durations, just that it exists and is returned in the order we expect.
+			for m := range ch {
+				metric := utils.ReadMetrics(m)
 				for _, suffix := range ignoredMetricSuffix {
 					if strings.Contains(metric.FqName, suffix) {
-						require.Equal(t, expectedMetric.FqName, metric.FqName)
 						continue metricsLoop
 					}
 				}
 
-				require.Equal(t, expectedMetric, metric)
+				metrics = append(metrics, metric)
 			}
-
+			assert.ElementsMatch(t, metrics, tt.expectedMetrics)
 		})
 	}
 }

--- a/pkg/google/gke/gke_test.go
+++ b/pkg/google/gke/gke_test.go
@@ -14,6 +14,7 @@ import (
 	billingv1 "cloud.google.com/go/billing/apiv1"
 	"cloud.google.com/go/billing/apiv1/billingpb"
 	"github.com/prometheus/client_golang/prometheus"
+	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 	computev1 "google.golang.org/api/compute/v1"
 	"google.golang.org/api/option"
@@ -193,6 +194,34 @@ func TestCollector_Collect(t *testing.T) {
 						"disk_type":        "persistent_volume",
 					},
 					Value:      0.15359342915811086,
+					MetricType: prometheus.GaugeValue,
+				},
+				{
+					FqName: "cloudcost_gcp_gke_persistent_volume_usd_per_hour",
+					Labels: map[string]string{
+						"cluster_name":     "test",
+						"namespace":        "cloudcost-exporter",
+						"persistentvolume": "test-ssd-disk",
+						"region":           "us-east4",
+						"project":          "testing-1",
+						"storage_class":    "pd-ssd",
+						"disk_type":        "persistent_volume",
+					},
+					Value:      0.15359342915811086,
+					MetricType: prometheus.GaugeValue,
+				},
+				{
+					FqName: "cloudcost_gcp_gke_persistent_volume_usd_per_hour",
+					Labels: map[string]string{
+						"cluster_name":     "test",
+						"namespace":        "cloudcost-exporter",
+						"persistentvolume": "test-disk",
+						"region":           "us-central1",
+						"project":          "testing-1",
+						"storage_class":    "pd-standard",
+						"disk_type":        "boot_disk",
+					},
+					Value:      0,
 					MetricType: prometheus.GaugeValue,
 				},
 				{
@@ -450,10 +479,7 @@ func TestCollector_Collect(t *testing.T) {
 			if len(metrics) == 0 {
 				return
 			}
-
-			for i, expectedMetric := range test.expectedMetrics {
-				require.Equal(t, expectedMetric, metrics[i])
-			}
+			assert.ElementsMatch(t, metrics, test.expectedMetrics)
 		})
 	}
 }

